### PR TITLE
chore(ci): add changesets

### PR DESCRIPTION
**Pull request by jeetiss**
### release automation with changesets

`changesets` provides an automated monopackage release flow with a review.

here are steps from PR to release:
- a user creates PR
- maintainer or user adds a changeset that contains a description and information about affected packages and release type (`patch`, `minor`, `major`) [like this](https://github.com/diegomura/react-pdf/pull/1581/files#diff-ec4a2ebf2bec9af4db2918086cd08d8c22115a12ee5e5e0155ab4d1d4c7d0901)
- GitHub action creates a [pull request](https://github.com/atlassian/changesets/pull/670) with generated changelogs and bumped versions for packages that are changed
- release engineer reviews changes and merge them
- GitHub action publishes the new versions to npm and creates a release note for all packages at github

Lerna has a similar release flow with semantic releases, but changeset is better because:
- files are more flexible for change descriptions than commit messages
- it has a review step that allows to check and change something at the last moment
- it has all steps automation out of the box

### TODOs

- [ ] add changesets bot
- [ ] add automation npm token for releases

### CODE REVIEW

**Comment by diegomura** [comment]
@jeetiss can you explain the motivation of this? Isn't Lerna do some of this stuff already?

**Comment by jeetiss** [comment]
@diegomura sure.
I'm finalizing our migration to v2, and after i'm back with motivation explanations and the new prs.

> now v2 works the same as v1, and i'm happy that i don't need to rewrite components

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+      - name: Create Release Pull Request or Publish to npm
+        uses: changesets/action@master
+        with:
+          # this expects you to have a script called release which does a build for your packages and calls changeset publish
+          publish: yarn release
+          version: yarn version-packages
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
```
Comment (jeetiss) [inline .github/workflows/release.yml:42]: @diegomura could you add `NPM_TOKEN` to github secrets? it should be an [automation token](https://github.blog/changelog/2020-10-02-npm-automation-tokens/) that github announce couple days ago

**Comment by jeetiss** [comment]
@diegomura i add some explanation to PR description

> Isn't Lerna do some of this stuff already?

yes, changesets is inspired by lerna but improves all aspects of release.

last thing. I tried to automate all possible things, so if you want to run some steps manually, just say and i prepare it here 😄

File: .github/workflows/release.yml (hunk 1 above)
Comment (jeetiss) [reply .github/workflows/release.yml]: `GITHUB_TOKEN` is provided on default, so you don't need to add it.

**Comment by diegomura** [comment]
@jeetiss thanks for merging this and improving the release cycle. And also for all the hard work you are putting on the project. I'll try to be more responsive and present than I've been for the last weeks. 

Can you quickly explain how this changes how to publish a new version? I'm not sure I grasped 100% of wht `changesets` does

**Comment by jeetiss** [comment]
@diegomura hooray 🙌🏻

I am almost ready with the new release, I'm going to finish [this fix] (https://github.com/diegomura/react-pdf/pull/1659) today, and after we can release

## changesets

I can't explain it better than changesets maintainers do, so read [this article](https://github.com/changesets/changesets/blob/main/docs/detailed-explanation.md)

here are the changesets of all PRs that I merged
- https://github.com/diegomura/react-pdf/tree/master/.changeset

changeset's github action generates a PR with generated changelogs and bumped versions for all packages, based on these files, you can check PR here
- https://github.com/diegomura/react-pdf/pull/1653

I have finished the setup, but you need to add `npm automation token` with the name `NPM_TOKEN` to github secrets, for automated publishing to npm, this is the last step for release

here the screen
Снимок экрана 2022-01-13 в 13 53 42

**Comment by diegomura** [comment]
I kept reading about it and things make much sense now. Thanks! Just added the token. Does that token allows changelogs to publish any package under my name? How is it that releases through GH will work?

**Comment by jeetiss** [comment]
thanks, it works ✨
https://github.com/diegomura/react-pdf/runs/4815869743?check_suite_focus=true#step:5:7

> Does that token allows changelogs to publish any package under my name? 

Yes, it can release all packages placed in this repository under your name.
It is secure, nobody can steal the token, and your secrets don't work in forks

> How is it that releases through GH will work?

it should work like a magic ✨

when I merge PR
- https://github.com/diegomura/react-pdf/pull/1653

all changeset files will be removed and release github action run `changeset publish` because of that, it should release the new packages automatically ✨

**Pull request merged**